openedx/core/lib/xblock_utils.py

```diff
@@ -294,7 +294,13 @@ def add_staff_markup(user, has_instructor_access, disable_staff_debug_info, bloc
         is_mongo_course = modulestore().get_modulestore_type(block.location.course_key) != ModuleStoreEnum.Type.xml
         is_studio_course = block.course_edit_method == "Studio"
 
-        if is_studio_course and is_mongo_course:
+        show_studio_link = True
+        try:
+            show_studio_link = user.instructor_dashboard_tabs.show_studio_link
```

Comment:
I think instead of try-except construction the better way for this block would be:
`show_studio_link = user.instructor_dashboard_tabs.show_studio_link if hasattr(user, 'instructor_dashboard_tabs') else True`